### PR TITLE
More datasets (QM7b, TU)

### PR DESCRIPTION
**Pull request by francescofarina**
## Proposed changes

Implementing more datasets.

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/datasets/dataset.py
Comment (francescofarina) [inline mlx_graphs/datasets/dataset.py:12]: Rethinking about it - shall we put it in the same folder where the code is being run? So that it's easier for anyone to explore it (rather than being in the mlx-graphs package repo inside a venv)

Comment (TristanBilot) [reply mlx_graphs/datasets/dataset.py]: Ok I just modified

**Comment by TristanBilot** [comment]
It's now possible to run tests for datasets with `pytest --datasets`.
Do you mind writing a few tests to verify my changes didn't break the qm7 dataset? Thx! :)

File: mlx_graphs/datasets/dataset.py
Comment (francescofarina) [inline mlx_graphs/datasets/dataset.py:34]: Why is `self.graphs` a `np.array`? I don't see any issues in making it a `list[GraphData]`

File: mlx_graphs/utils/array_ops.py
Comment (francescofarina) [inline mlx_graphs/utils/array_ops.py:105]: This should probably stay in `transformations.py` and the input should be validated.

Comment (francescofarina) [reply mlx_graphs/utils/array_ops.py]: moved in [9b1ec84](https://github.com/mlx-graphs/mlx-graphs/pull/81/commits/9b1ec8490e9eacdee309a149f75197bbebf2f9a4)

**Review by TristanBilot — APPROVED** [review]

**Comment by TristanBilot** [comment]
I'm waiting for your last reviews before merging ✅

**Comment by francescofarina** [comment]
I fixed a couple of things, nothing major. There are a couple of type check errors here and there but nothing to worry about for the moment. We may want to make the `edge_index` arg of `GraphData` not optional to fix most of them. But amazing overall - let's merge!

**Pull request merged**
